src/pip/_internal/operations/install/wheel.py

```diff
@@ -583,8 +583,28 @@ def data_scheme_file_maker(zip_file, scheme):
         def make_data_scheme_file(record_path):
             # type: (RecordPath) -> File
             normed_path = os.path.normpath(record_path)
-            _, scheme_key, dest_subpath = normed_path.split(os.path.sep, 2)
-            scheme_path = scheme_paths[scheme_key]
+            try:
+                _, scheme_key, dest_subpath = normed_path.split(os.path.sep, 2)
+            except ValueError:
+                message = (
+                    "Unexpected file in {}: {!r}. .data directory contents"
+                    " should be named like: '<scheme key>/<path>'."
+                ).format(wheel_path, record_path)
+                raise InstallationError(message)
+
+            try:
+                scheme_path = scheme_paths[scheme_key]
+            except KeyError:
+                valid_scheme_keys = ", ".join(scheme_paths.keys())
```

Comment:
The `keys()` call here is unnecessary, but I guess it wouldn’t hurt to be explicit. Maybe it’d be a good idea to sort the keys so the output is consistent though?

Reply:
I forgot that dicts act as an iterable of their keys by default, thank you. I applied both suggestions in 3f9b326.